src/tests/presets/test_account.cairo

```diff
@@ -0,0 +1,454 @@
+use openzeppelin::account::AccountComponent::{OwnerAdded, OwnerRemoved};
+use openzeppelin::account::AccountComponent::{TRANSACTION_VERSION, QUERY_VERSION};
+use openzeppelin::account::interface::ISRC6_ID;
+use openzeppelin::account::{AccountABIDispatcherTrait, AccountABIDispatcher};
+use openzeppelin::introspection::interface::ISRC5_ID;
+use openzeppelin::presets::Account;
+use openzeppelin::tests::account::test_account::{
+    deploy_erc20, SIGNED_TX_DATA, SignedTransactionData
+};
+use openzeppelin::tests::utils::constants::{PUBKEY, NEW_PUBKEY, SALT, ZERO, RECIPIENT};
+use openzeppelin::tests::utils;
+use openzeppelin::token::erc20::interface::{IERC20DispatcherTrait, IERC20Dispatcher};
+use openzeppelin::utils::selectors;
+use openzeppelin::utils::serde::SerializedAppend;
+use starknet::ContractAddress;
+use starknet::account::Call;
+use starknet::contract_address_const;
+use starknet::testing;
+
+fn CLASS_HASH() -> felt252 {
+    Account::TEST_CLASS_HASH
+}
+
+//
+// Setup
+//
+
+fn setup_dispatcher() -> AccountABIDispatcher {
+    let mut calldata = array![PUBKEY];
+    let target = utils::deploy(CLASS_HASH(), calldata);
+    utils::drop_event(target);
+
+    AccountABIDispatcher { contract_address: target }
+}
+
+fn setup_dispatcher_with_data(data: Option<@SignedTransactionData>) -> AccountABIDispatcher {
+    testing::set_version(TRANSACTION_VERSION);
+
+    let mut calldata = array![];
+    if data.is_some() {
+        let data = data.unwrap();
+        testing::set_signature(array![*data.r, *data.s].span());
+        testing::set_transaction_hash(*data.transaction_hash);
+
+        calldata.append(*data.public_key);
+    } else {
+        calldata.append(PUBKEY);
+    }
+    let address = utils::deploy(CLASS_HASH(), calldata);
+    AccountABIDispatcher { contract_address: address }
+}
+
+//
+// constructor
+//
+
+#[test]
+#[available_gas(2000000)]
+fn test_constructor() {
+    let mut state = Account::contract_state_for_testing();
+    Account::constructor(ref state, PUBKEY);
+
+    assert_only_event_owner_added(PUBKEY, ZERO());
+
+    assert(Account::PublicKeyImpl::get_public_key(@state) == PUBKEY, 'Should return PUBKEY');
```

Comment:
small detail, but this is inconsistent with the component's initializer test which reads:

```js
    assert(state.account.get_public_key() == PUBKEY, 'Should return public key');
```

it's similar with the `ISRC5/6` error messages below, they're not normalized across the codebase (e.g. account component) and they probably should. some day 😅

Reply:
Updated

Reply:
although this specific suggestion got accepted, we still need to normalize. i [created an issue](https://github.com/OpenZeppelin/cairo-contracts/issues/820), we can discuss there if we agree the normalization should be done through constants instead of copying the string each time. bear in mind this is low prio as it's not user facing